gui/gui/src/TGMenu.cxx

```diff
@@ -1249,10 +1249,18 @@ void TGPopupMenu::PlaceMenu(Int_t x, Int_t y, Bool_t stick_mode, Bool_t grab_poi
    // Parent is root window for a popup menu
    gVirtualX->GetWindowSize(fParent->GetId(), rx, ry, rw, rh);
 
-   if (x < 0) x = 0;
-   if (x + fMenuWidth > rw) x = rw - fMenuWidth;
-   if (y < 0) y = 0;
-   if (y + fMenuHeight > rh) y = rh - fMenuHeight;
+   if (gVirtualX->InheritsFrom("TGWin32")) {
+      if ((x > 0) && ((x + abs(rx) + (Int_t)fMenuWidth) > (Int_t)rw))
```

Comment:
@bellenot You do not check here negative `x` and `y` values. Is it intended?

Reply:
yes, because when it is negative, the position is correct...